Gems/Atom/Feature/Common/Code/Source/Mesh/MeshFeatureProcessor.cpp

```diff
@@ -163,11 +165,77 @@ namespace AZ
         void MeshFeatureProcessor::OnBeginPrepareRender()
         {
             m_meshDataChecker.soft_lock();
+
+            // Must read cvar from AZ::Console due to static variable in multiple libraries, see ghi-5537
+            bool enablePerMeshShaderOptionFlagsCvar = false;
+            if (auto* console = AZ::Interface<AZ::IConsole>::Get(); console != nullptr)
+            {
+                console->GetCvarValue("r_enablePerMeshShaderOptionFlags", enablePerMeshShaderOptionFlagsCvar);
+            }
+
+            if (m_enablePerMeshShaderOptionFlags && !enablePerMeshShaderOptionFlagsCvar)
+            {
+                // Per mesh shader option flags was on, but now turned off, so reset all the shader options.
+                for (auto& model : m_modelData)
+                {
+                    for (RPI::MeshDrawPacketList& drawPacketList : model.m_drawPacketListsByLod)
+                    {
+                        for (RPI::MeshDrawPacket& drawPacket : drawPacketList)
+                        {
+                            model.m_cullable.m_flags = 0;
+                            drawPacket.ClearShaderOptions();
```

Comment:
This won't just clear the light options, it will also clear anything else that was set via MeshDrawPacket::SetShaderOption.  For example, see o_meshUseForwardPassIBLSpecular. Is there something that reapplies those options?

Reply:
Going to add an UnsetShaderOption() so that values can be cleared (and returned to whatever the default state in the shader is).